File: ios/src/ReactNative/RNIContextMenuView/RNIContextMenuViewManager.m

//
//  RCTContextMenuManager.m
//  nativeUIModulesTest
//
//  Created by Dominic Go on 7/14/20.
//

#import <React/RCTViewManager.h>
 

@interface RCT_EXTERN_MODULE(RNIContextMenuViewManager, RCTViewManager)

// MARK: - Export Props - Events
// -----------------------------

RCT_EXPORT_VIEW_PROPERTY(onMenuWillShow  , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMenuWillHide  , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMenuWillCancel, RCTBubblingEventBlock);

RCT_EXPORT_VIEW_PROPERTY(onMenuDidShow  , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMenuDidHide  , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onMenuDidCancel, RCTBubblingEventBlock);

RCT_EXPORT_VIEW_PROPERTY(onPressMenuItem   , RCTBubblingEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onPressMenuPreview, RCTBubblingEventBlock);

RCT_EXPORT_VIEW_PROPERTY(onMenuWillCreate, RCTBubblingEventBlock);

// MARK: - Export Props - Values
// -----------------------------

RCT_EXPORT_VIEW_PROPERTY(menuConfig   , NSDictionary);
RCT_EXPORT_VIEW_PROPERTY(previewConfig, NSDictionary);

// MARK: Export View Manager Commands
// ----------------------------------

RCT_EXTERN_METHOD(dismissMenu:(nonnull NSNumber *)node);

@end

File: ios/RNIContextMenuView/RNIContextMenuViewManager.swift
//
//  RCTContextMenuManager.swift
//  nativeUIModulesTest
//
//  Created by Dominic Go on 7/14/20.
//

import Foundation

@objc(RNIContextMenuViewManager)
class RNIContextMenuViewManager: RCTViewManager {
  
  override static func requiresMainQueueSetup() -> Bool {
    return true;
  };
  
  override func view() -> UIView! {
    if #available(iOS 13.0, *) {
      return RNIContextMenuView(bridge: self.bridge);
      
    } else {
      return RCTView();
    };
  };
  
  @objc func dismissMenu(_ node: NSNumber) {
    DispatchQueue.main.async {
      guard #available(iOS 13.0, *),
            let view = self.bridge.uiManager.view(forReactTag: node),
            let contextMenuView = view as? RNIContextMenuView
      else { return };
      
      contextMenuView.dismissMenu();
    };
  };
};
